float64[] initial_state
float64[] target_state
